client/components/App.jsx

```diff
@@ -76,45 +40,89 @@ class App extends React.Component {
       .then(res => {
         return res.json();
       })
-      .then(({reviewData, starData, featureData}) => {
-        let update = this.state.allReviews;
-        update[productId] = reviewData;
-        this.setState({
-          starData,
-          featureData,
-          allReviews: update,
-        }, err => {
-          if (err) {
-            console.log('Error in setState');
-            throw err;
-          }
-        });
+      .then(reviews => {
+        this.processReviews(reviews);
       })
       .catch(err => {
         console.log('ERROR FETCHING REVIEWS****************', err);
       });
   }
 
-  
+  processReviews({reviewData, starData, featureData}) {
+    let mediaList = buildMediaList(reviewData);
+    this.setState({
+      reviewData,
+      starData,
+      featureData,
+      mediaList
+    }, err => {
+      if (err) {
+        console.log('Error in setState');
+        throw err;
+      }
+    });
+  }
+
+  displayImageInModal(e) {
+    e.preventDefault();
+    let data = e.target.dataset;
+    this.setState({
+        modal: {
+          display: true,
+          mediaIndex: parseInt(data.mediaIndex),
+          reviewIndex: parseInt(data.reviewIndex)
+        }
+    }, err => {
+      if (err) throw err;
+    });
+  }
+
+  toggleModal() {
+    return (e) => {
+      e.preventDefault();
+      this.setState({
+        modal: {
+          display: !this.state.modal.display,
```

Comment:
Also I'm just curious about this, but I see that `toggleModal` returns a function and so in **line 105** you assign to the `ImageModal` component `toggleModal={this.toggleModal()}`, which is the function that invoking `toggleModal` returns. Is there a reason for doing this instead of having the function `toggleModal` accept a parameter (so for **line 80** to be `toggleModal(e)`)?  You could then delete **line 81 and 92**, and in **line 105** you can assign `ImageModal`'s property to be `toggleModal=(this.toggleModal.bind(this)}`?

Reply:
Does this have to do with Alex's comment on Slack about avoinding having to bind this explicitly?

Reply:
Hey JJ, thanks again for reviewing my PR!

toggleModal doesn't explicitly set the state to `false` because while it is usually used to close the modal, in one case (clicking 'See all customer images') it's used to launch the modal.

Yeah, I wanted to try out what Alex said. I haven't figured out which method I prefer yet, though.